src/app/sessions/components/session-table/session-table.component.ts

```diff
@@ -40,43 +42,76 @@ export class SessionTableComponent implements OnChanges {
 
     this.tableVisible = false;
 
-    this.dataSource = new MatTableDataSource(this.sessions);
-  }
-
-  parseDate(date) {
-      return moment(date).format('DD/MM/YYYY');
+    this.dataSource = new MatTableDataSource(this.decorateSessions(this.sessions));
   }
 
   parseTime(date: moment.Moment) {
     return date.format('HH:mm');
   }
 
   humanizeDuration(duration) {
-      return moment.duration(duration).humanize();
+      return moment.utc(moment.duration(duration).asMilliseconds()).format('HH:mm');
   }
 
-  calculateUtilized(duration: string, allocated: moment.Duration) {
+  calculateUtilized(duration: number, allocated: moment.Duration): number {
     return this.sessionsStatsService.calculateUtilizedDuration(moment.duration(duration), allocated);
   }
 
-  calculateAllocated(session) {
+  calculateAllocated(session: SessionViewModel) {
     return this.sessionsStatsService.calculateAllocatedHearingsDuration(session);
   }
 
-  calculateAvailable(duration: string, allocated: moment.Duration) {
+  calculateAvailable(duration: number, allocated: moment.Duration) {
     return this.sessionsStatsService.calculateAvailableDuration(moment.duration(duration), allocated);
   }
 
-  toggleSession(session) {
+  toggleSession(session: SessionViewModel) {
     this.selectedSesssion.toggle(session);
     this.selectSession.emit(this.selectedSesssion.isSelected(session) ? session : {})
   }
 
   ngOnChanges() {
       if (this.sessions) {
           this.tableVisible = true;
-          this.dataSource = new MatTableDataSource(this.sessions);
+          this.dataSource = new MatTableDataSource(this.decorateSessions(this.sessions));
+
+          this.dataSource.sortingDataAccessor = (item, property) => {
+              switch (property) {
+                  case 'person':
+                      return getPropertyMemberOrNull(item, property, 'name');
+                  case 'room':
+                      return getPropertyMemberOrNull(item, property, 'name');
+                  case 'hearingParts':
+                      return getPropertyMemberOrNull(item, property, 'length');
+                  case 'time':
+                  case 'date':
+                      return item['start'].unix();
+                  case 'duration':
+                      return moment.duration(item[property]);
+                  default:
+                      return item[property];
+              }
+          }
+
+          this.dataSource.sort = this.sort;
       }
   }
 
+    private decorateSessions(sessions: SessionViewModel[]) {
```

Comment:
maybe that calculation should be moved to src/app/sessions/reducers/index.ts?

Reply:
Try to move the calculation to session.reducer.ts